ansible_collections/arista/cvp/plugins/module_utils/device_tools.py

```diff
@@ -741,75 +896,25 @@ def manager(self, user_inventory: DeviceInventory, search_mode: str = FIELD_HOST
-                \nMake sure that the devices are provisioned and defined with the full fqdn name \
-                (including the domain name) if needed.'.format(str(list_non_existing_devices))
-            MODULE_LOGGER.error(error_message)
-            self.__ansible.fail_json(msg=error_message)
-
-        # Need to collect all missing device systemMacAddress
-        # deploy needs to locate devices by mac-address
-        if self.__search_by == FIELD_FQDN:
-            user_inventory = self.refresh_systemMacAddress(user_inventory=user_inventory)
-
-        elif self.__search_by == FIELD_HOSTNAME:
-            user_inventory = self.refresh_systemMacAddress(user_inventory=user_inventory)
-
-        elif self.__search_by == FIELD_SERIAL:
-            user_inventory = self.refresh_fqdn(user_inventory=user_inventory)
-            user_inventory = self.refresh_systemMacAddress(user_inventory=user_inventory)
-
-        action_result = self.deploy_device(user_inventory=user_inventory)
-        if action_result is not None:
-            for update in action_result:
-                cv_deploy.add_change(change=update)
-
-        action_result = self.move_device(user_inventory=user_inventory)
-        if action_result is not None:
-            for update in action_result:
-                cv_move.add_change(change=update)
+        if state == 'present':
+            MODULE_LOGGER.info('Processing data to create/update devices')
+            response = self.__state_present(user_inventory=user_inventory, apply_mode=apply_mode)
 
-        action_result = self.apply_configlets(user_inventory=user_inventory)
-        if action_result is not None:
-            for update in action_result:
-                cv_configlets_attach.add_change(change=update)
-
-        if apply_mode == 'strict':
-            action_result = self.detach_configlets(
-                user_inventory=user_inventory)
-            if action_result is not None:
-                for update in action_result:
-                    cv_configlets_detach.add_change(change=update)
-
-        response.add_manager(cv_move)
-        MODULE_LOGGER.debug('AnsibleResponse updated, new content with cv_move: %s', str(response.content))
-        response.add_manager(cv_deploy)
-        MODULE_LOGGER.debug('AnsibleResponse updated, new content with cv_deploy: %s', str(response.content))
-        response.add_manager(cv_configlets_attach)
-        MODULE_LOGGER.debug('AnsibleResponse updated, new content with cv_configlets_attach: %s', str(response.content))
-        response.add_manager(cv_configlets_detach)
-        MODULE_LOGGER.debug('AnsibleResponse updated, new content with cv_configlets_detach: %s', str(response.content))
+        elif state == 'absent':
+            MODULE_LOGGER.info('Processing data to reset devices')
+            response = self.__state_absent(user_inventory=user_inventory)
```

Comment:
IMHO, 'absent' is not a good keyword to initiate the factory reset of the device. 
People might use `absent` thinking that it will just remove the device from provisioning, and thus initiate the factory reset instead. That might lead to potential outages in the network
To be safe, I would change this keyword to `factory_reset` so there is no ambiguity, event if we do not have any `absent` state as of now in the collection

My 2 cents about this...

Reply:
Changed to `factory_reset` as discussed.